mathics/core/rules.py

```diff
@@ -259,14 +259,25 @@ def __repr__(self) -> str:
 
     def __getstate__(self):
         odict = self.__dict__.copy()
-        del odict["function"]
-        odict["function_"] = (self.function.__self__.get_name(), self.function.__name__)
+        # If function is a method of a builtin class
+        # store the name of the class and the name
+        # of the method instead of the callable.
```

Comment:
If this is being done to make the definition pickle-able,  please state that here.

Reply:
Check if this makes more sense.

Reply:
The comment is repeating what is done. But the _why_ is missing.  

There is much bravado over the benefit and what it provides, but the place where the crux of the change happens, there was no mention for the reason behind the change.

When one changes code for a very specific or important reason, we should state what the reason is at the place of change. 

Otherwise, someone may just revert the change because it seems complicated and the reason is not given.

Reply:
Interesting. 
Maybe you can help me to improve the comment. The reason for the change is that at some point, we allow to the `Builtin.contribute` method to create rules which instead of calling an eval method, called an internal `unavailable_function` defined in the line 456 of the `mathics.builtin.base` module. The goal of this function is to show a message telling the user that the method is not available because a certain library is not available in the system.

This breaks the pickle mechanism, so here I proposed to replace it with a callable object, that essentially does the same but accepts to be pickled. 

Then we have two possibilities for the content of the `function` attribute: either it is a method of a `Builtin` class, or is a callable object, which can be pickled. If it is a method, then the code makes the same that it did before the PR, but if it another kind of callable, it just stores it.

In any case, if someone change this to a simpler form, and the new serialization test passes, everything would be OK.

Reply:
Give me some time and I will happily add/revise the comment and you can review. 

There is a lot of information in these  PR  comments/discussion (in a not compact or even rambling form). 

The goal I think is basically to say what has been said here inside the code somehow.